gateway/src/apicast/proxy.lua

```diff
@@ -206,36 +208,29 @@ function _M.set_upstream(service)
   ngx.req.set_header('Host', upstream.host or ngx.var.host)
 end
 
------
--- call the proxy and return a handler function
--- that will perform an action based on the path and backend version
--- @tparam service service service object
--- @treturn nil|function access function (when the request needs to be authenticated with this)
--- @treturn nil|function handler function (when the request is not authenticated and has some own action)
-function _M:call(service)
-  service = _M.set_service(service or ngx.ctx.service)
-
-  self.oauth = service:oauth()
+function _M:handle_oauth(service)
+  local oauth = service:oauth()
+  self.oauth = oauth
```

Comment:
Maybe this assignment is not needed then.

Reply:
It is, unless we change something.
There's `self.oauth:transform_credentials` at the end of `rewrite()`.
We should check if it's possible to move that block of code to `handle_oauth`.

Reply:
Yep. It was needed for transform_credentials. Now it returns `oauth` and is assigned in the `:rewrite`.

Reply:
Better :+1: